Sources/MapboxCoreNavigation/AlternativeRoutesCenter.swift

```diff
@@ -0,0 +1,135 @@
+import Foundation
+import MapboxDirections
+import MapboxNavigationNative
+
+/**
+ `AlternativeRoutesCenter` observer to track alternative routes updates.
+ */
+public protocol NavigatorAlternativesStoreDelegate: AnyObject {
+    /**
+     Called when center has detected a change in alternative routes list.
+     
+     - parameter center: `AlternativeRoutesCenter` reporting an update
+     - parameter didReportNewAlternatives: Indicies within `AlternativeRoutesCenter.alternatives` array where new alternative routes are appended.
+     - parameter removedAlternatives: Array of alternative routes which are no longer actual.
+     */
+    func alternativeRoutesCenter(_ center: AlternativeRoutesCenter, didReportNewAlternatives: IndexSet, removedAlternatives: [AlternativeRoute])
+    
+    /**
+     Called when center has failed to  change alternative routes list.
+     
+     - parameter center: `AlternativeRoutesCenter` reporting an update
+     - parameter didFailToUpdateAlternatives: An error occured.
+     */
+    func alternativeRoutesCenter(_ center: AlternativeRoutesCenter, didFailToUpdateAlternatives: AlternativeRouteError)
+}
+
+/**
+ Provides notifications and access to `AlternativeRoute`s found during navigation.
+ */
+public class AlternativeRoutesCenter {
```

Comment:
We should replace AlternativeRoutesCenter, NavigatorAlternativesStoreDelegate, and AlternativeRoutesObserver with notifications coming out of NotificationCenter.

The naming difficulty for these three types stems from using a tightly coupled many-delegate pattern, which doesn’t really have much precedent in Cocoa frameworks.

Moreover, strongly holding onto observers could lead to retain cycles and memory leaks. After all, AlternativeRoutesCenter is owned by RouteController, which is ultimately owned by NavigationViewController. But what if NavigationViewController wants to know when there are alternative routes? It would register as an observer, causing it to be held strongly by AlternativeRoutesCenter.

Notifications would allow us to minimize the number of moving parts. If some UI class only wants to know about alternatives for a particular main route, it can observe for notifications that have that route as the object. But most likely its lifecycle would line up with that of the RouteController anyways.

Reply:
The reason I decided to use a special "store/center" object is because use should be able not only to receive alternative routes, but also manipulate (remove) it. This would be hard to achieve using notifications. Another option I was thinking is have `AlternativeRoutesCenter` as a singleton which does not belong to any particular controller, but I didn't want to add one more singleton to the SDK and also alternatives exist only during AG session, so I've decided to include it into RouteController.

Reply:
How about separating the notifying functionality from the alternative route management functionality? UI code, for instance, could be notified about alternative route detections via notifications (which we always implement in parallel to delegate protocols), but random access and manipulation of alternative routes goes through RouteController or, if you like, a subsidiary AlternativeRoutesCenter class wholly owned by RouteController. (At this point, I’m not sure why it would need to be a separate class though.)